Sources/Datadog/Datadog.swift

```diff
@@ -154,22 +163,23 @@ public class Datadog {
     /// - Parameters:
     ///   - extraInfo: User's additionall custom attributes
     public static func addUserExtraInfo(
-        _ extraInfo: [AttributeKey: AttributeValue?]
+        _ extraInfo: [AttributeKey: AttributeValue?],
+        in core: DatadogCoreProtocol = CoreRegistry.default
     ) {
-        let core = CoreRegistry.default as? DatadogCore
+        let core = core as? DatadogCore
         core?.addUserExtraInfo(extraInfo)
     }
 
     /// Sets the tracking consent regarding the data collection for the Datadog SDK.
     /// - Parameter trackingConsent: new consent value, which will be applied for all data collected from now on
-    public static func set(trackingConsent: TrackingConsent) {
-        let core = CoreRegistry.default as? DatadogCore
+    public static func set(trackingConsent: TrackingConsent, in core: DatadogCoreProtocol = CoreRegistry.default) {
```

Comment:
question/ Do we do it the same way in `dd-sdk-android`? Static API that queries instance under the hood, not instance API that requires user to query the instance? Both are fine, I'm just making sure we don't diverge 👍.

Reply:
I can't find it in Android, but this one complies with the convention. @0xnm How do you set the consent per core?

Reply:
yes, we set consent per core. In general, all the methods like `setTrackingConsent`, `setUserInfo`, etc. should be called on the instance directly, meaning the syntax will be `Datadog.getInstance().setTrackingConsent`.

The reason for that is that we are going to expose `Core` interface to the customer side (it is the instance returned by the `Datadog.getInstance`), and if we add a static method to set such things, than we would have both `Datadog.setTrackingConsent` and `Datadog.getInstance().setTrackingConsent` to do the same, which brings some ambiguity.

Reply:
Thanks, @0xnm! The core interface does not expose consent and user info apis, it should be tackled in dedicated PR.

Reply:
Everything declared [here](https://github.com/DataDog/dd-sdk-ios/blob/develop/Sources/Datadog/DatadogInternal/DatadogCoreProtocol.swift#L16). 

This `sdkInstance` is meant to pass the instance to static methods, like:
```swift
Logs.enable(in: Datadog.sdkInstance(named: "custom-instance"))
```
This variant allow to do this:
```swift
Datadog.set(trackingConsent: .granted)
// or
Datadog.set(trackingConsent: .granted, in: Datadog.sdkInstance(named: "custom-instance"))
```

Which complies with the approach we've discussed and aligns with other static methods across modules.

Reply:
@maxep @0xnm what is the consensus here? I consider this a blocker unless we align.

Reply:
We currently can't align without changing the `DatadogCoreProtocol`, I consider this outside of the scope of this PR.
We have [RUMM-3295](https://datadoghq.atlassian.net/browse/RUMM-3295) to align 👍

Reply:
👌 sounds good, thanks 🙂 !